cli/pkg/digger/digger.go

```diff
@@ -359,38 +360,33 @@ func run(command string, job orchestrator.Job, policyChecker policy.Checker, org
 			return nil, comment, fmt.Errorf(comment)
 		} else {
 
-			// Running plan before apply to check against access policy
-			// TODO only run plan if policy check present. Always running plan is slow.
+			// checking policies (plan, access)
 
-			_, planPerformed, isNonEmptyPlan, _, planJsonOutput, err := diggerExecutor.Plan()
+			storedPlanJson, err := retrievePlanBeforeApply(planStorage, planPathProvider, diggerExecutor)
+			var planPolicyViolations []string
 			if err != nil {
```

Comment:
I don't think we should check for error and proceed regardless, an error retriving plan should abort immediately, there are many reasons for failing to retrive a plan and we are skipping those

We should instead check explicitly that a Noop PlanStorageProvider and in THAT case to skip the violation checks. For now we can check for os.GetEnv("PLAN_UPLOAD_DESTINATION") == "" then skip

and move the retrival to the else part